File: 0255_Problem_1.rb
# Problem Link: https://leetcode.com/contest/weekly-contest-255/problems/find-greatest-common-divisor-of-array/

# Solution 1 (original)
def find_gcd(nums)
    return nums.max.gcd(nums.min)
end

# Solution 2 (omit return)
def find_gcd(nums)
    nums.max.gcd(nums.min)
end

# Solution 3 (use minmax)
def find_gcd(nums)
    nums.minmax.reduce(:gcd)
end

File: 0255_Problem_1.jl
# Problem Link: https://leetcode.com/contest/weekly-contest-255/problems/find-greatest-common-divisor-of-array/

# Solution 1 (original)
function find_gcd(nums)
    gcd(min(nums...), max(nums...))
end

# Solution 2 (assignment form)
find_gcd(nums) = gcd(min(nums...), max(nums...))

# Solution 3 (use minimum/maximum)
find_gcd(nums) = gcd(minimum(nums), maximum(nums))
            
# Soluiton 4 (use extrema)
find_gcd(nums) = gcd(extrema(nums)...)

# Solution 5 (POINT FREE!)
find_gcd = gcd ∘ collect ∘ extrema
